src/main/webapp/app/shared/vertical-progress-bar/vertical-progress-bar.component.ts

```diff
@@ -0,0 +1,108 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, HostBinding, Input, OnInit } from '@angular/core';
+
+/**
+ * Simple Vertical Progress Bar without any external dependencies
+ * <ul>
+ *     <li> Supports three different colors depending on fill level
+ *     <li> [upperBorder, 100] -> upper color
+ *     <li> (lowerBorder, upperBorder) -> intermediate color
+ *     <li> [0,lowerBorder] -> lower color
+ * </ul>
+ */
+@Component({
+    selector: 'jhi-vertical-progress-bar',
+    templateUrl: './vertical-progress-bar.component.html',
+    styleUrls: ['./vertical-progress-bar.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VerticalProgressBarComponent implements OnInit {
+    // CSS VARIABLES START
+    @HostBinding('style.--progress-bar-height')
+    heightCSS = '50px';
+    @HostBinding('style.--progress-bar-width')
+    widthCSS = '50px';
+    @HostBinding('style.--fill-level')
+    fillLevelCSS = '0%';
+    @HostBinding('style.--fill-color')
+    fillColorCSS = 'green';
+    @HostBinding('style.--fill-duration')
+    fillDurationCSS = '1s';
+    @HostBinding('style.--border-radius')
+    borderRadiusCSS = '16px';
+    // CSS VARIABLES END
+
+    constructor(private cdr: ChangeDetectorRef) {}
+
+    ngOnInit(): void {
+        this.setFillColor();
+    }
+
+    fillLevelInPercentInternal = 0;
+    lowerBorderInternal = 60;
+    upperBorderInternal = 90;
+
+    @Input()
+    set lowerBorder(border: number) {
+        this.lowerBorderInternal = Math.round(Math.max(border, 0));
+        this.setFillColor();
+        this.cdr.markForCheck;
+    }
+
+    @Input()
+    set upperBorder(border: number) {
+        this.upperBorderInternal = Math.round(Math.min(border, 100));
+        this.setFillColor();
+        this.cdr.markForCheck;
+    }
+    @Input()
+    lowerColor = 'green';
+    @Input()
+    intermediateColor = 'orange';
+    @Input()
+    upperColor = 'red';
```

Comment:
These colors are not theme-aware, hard-coded colors. Please use default colors from the theme, in this case that would be ``success``, ``warning`` and ``danger``. CSS variables for these colors are globally available

Reply:
Done in 491adfb2516522167fc956c9433b5ebc9e81954e